api/apps/v1alpha1/nemo_datastore_types.go

```diff
@@ -157,99 +203,306 @@ func (n *NemoDatastore) GetStandardLabels() map[string]string {
+		},
+		{
+			Name:  "GITEA__DATABASE__HOST",
+			Value: fmt.Sprintf("%s:%d", dbSetting.Host, dbSetting.Port),
+		},
+		{
+			Name:  "GITEA__DATABASE__USER",
+			Value: dbSetting.Credentials.User,
+		},
 	}
+	return envVars
 }
 
-func (n *NemoDatastore) GetInitContainers() []corev1.Container {
-	image := n.Spec.DataStoreParams.InitContainerImage
-	if image == "" {
-		image = "busybox"
-	}
-	cmd := n.Spec.DataStoreParams.InitContainerCommand
-	if len(cmd) == 0 {
-		cmd = []string{
-			"sh",
-			"-c",
-			fmt.Sprintf("until nc -z %s %s; do echo \"PostgreSQL is unavailable. Sleeping for 5 seconds\"; sleep 5; done;", n.Spec.DataStoreParams.DatabaseHost, n.Spec.DataStoreParams.DatabasePort),
-		}
-	}
-	return []corev1.Container{
+// GetVolumes returns volumes for the NemoDatastore container
+func (n *NemoDatastore) GetVolumes() []corev1.Volume {
+	/*volumes:
+	  - name: init
+	    secret:
+	      defaultMode: 110
+	      secretName: datastore-nemo-datastore-init
+	  - name: config
+	    secret:
+	      defaultMode: 110
+	      secretName: datastore-nemo-datastore
+	  - name: inline-config-sources
+	    secret:
+	      defaultMode: 420
+	      secretName: datastore-nemo-datastore-inline-config
+	  - emptyDir: {}
+	    name: temp
+	  - name: data
+	    persistentVolumeClaim:
+	      claimName: datastore-shared-storage
+	*/
+	var initMode = int32(110)
+	var configMode = int32(420)
+
+	volumes := []corev1.Volume{
 		{
-			Name:    "wait-postgres-ready",
-			Image:   image,
-			Command: cmd,
+			Name: "init",
+			VolumeSource: corev1.VolumeSource{
+				Secret: &corev1.SecretVolumeSource{
+					SecretName:  n.Spec.Secrets.DataStoreInitSecret,
+					DefaultMode: &initMode,
+				},
+			},
+		},
+		{
+			Name: "config",
+			VolumeSource: corev1.VolumeSource{
+				Secret: &corev1.SecretVolumeSource{
+					SecretName:  n.Spec.Secrets.DataStoreConfigSecret,
+					DefaultMode: &initMode,
+				},
+			},
+		},
+		{
+			Name: "inline-config-sources",
+			VolumeSource: corev1.VolumeSource{
+				Secret: &corev1.SecretVolumeSource{
+					SecretName:  n.Spec.Secrets.DataStoreInlineConfigSecret,
+					DefaultMode: &configMode,
+				},
+			},
+		},
+		{
+			Name: "temp",
+			VolumeSource: corev1.VolumeSource{
+				EmptyDir: &corev1.EmptyDirVolumeSource{},
+			},
 		},
 	}
+
+	if n.Spec.PVC != nil {
+		volumes = append(volumes, corev1.Volume{
+			Name: "data",
+			VolumeSource: corev1.VolumeSource{
+				PersistentVolumeClaim: &corev1.PersistentVolumeClaimVolumeSource{
+					ClaimName: n.GetPVCName(),
+				},
+			},
+		})
+	} else {
+		volumes = append(volumes, corev1.Volume{
+			Name: "data",
+			VolumeSource: corev1.VolumeSource{
+				EmptyDir: &corev1.EmptyDirVolumeSource{},
+			},
+		})
+	}
+	return volumes
+}
+
+func (n *NemoDatastore) ShouldCreatePersistentStorage() bool {
+	return n.Spec.PVC != nil && n.Spec.PVC.Create != nil && *n.Spec.PVC.Create
+}
+
+// GetStandardAnnotations returns default annotations to apply to the NemoDatastore instance
```

Comment:
Fix the method comment.

Reply:
Remove the function as it's no longer used